src/main/java/com/nique1/musicmanagementsystem/song/SongServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.nique1.musicmanagementsystem.song;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class SongServiceImpl implements SongService{
+    private final SongRepository songRepository;
+    @Autowired
+    public SongServiceImpl(SongRepository songRepository) {
+        this.songRepository = songRepository;
+    }
+
+    @Override
+    public List<SongEntity> getSongsById(int id) {
+        List<SongEntity> songs =  songRepository.findSongsBySongId(id).stream().toList();
+        if(songs.isEmpty()){
+            throw new IllegalArgumentException("Invalid song id" + id);
```

Comment:
Aaaa, tutaj to obsługujesz - ale to spowoduje błąd 500, a tymczasem sytuacja w której klient prosi o piosenkę o id który nie jest znany to problem po jego stronie i nie sugeruje błędu w logice aplikacji.

Tak naprawdę ta metoda powinna zwracać pojedynczy element `Optional<SongEntity>` i w kontrolerze rozpakowywać go lub rzucać `ResponseStatusException` z odpowiednim kodem błędu.